cmd/m3db-operator/main.go

```diff
@@ -83,35 +93,70 @@ func main() {
 		logger.Fatal("failed to create k8sclient", zap.Error(err))
 	}
 
+	stopCh := make(chan struct{})
+
+	// TODO(schallert): move these to k8sops client once we're confident we want
+	// them
+	//
+	// TODO(schallert): not sure if we need podinformers as well or if
+	// abstractions of statefulsets will do
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, _informerSyncDuration)
+	m3dbClusterInformerFactory := informers.NewSharedInformerFactory(crdClient, _informerSyncDuration)
+
+	opts := []controller.Option{
+		controller.WithKubeInformerFactory(kubeInformerFactory),
+		controller.WithM3DBClusterInformerFactory(m3dbClusterInformerFactory),
+		controller.WithLogger(logger),
+		controller.WithKClient(k8sclient),
+		controller.WithCRDClient(crdClient),
```

Comment:
So what was the verdict on placing the clients within the controller itself? Should we remove the k8sops interface completely? It might make testing a bit more hairy though.

Reply:
I think my vote would be to wait until we have an idea of the full set of APIs we're going to need and then create an abstraction after that. Thoughts?

Re: the testing, it seems like the things that k8sops abstracts all have mock / fake packages

Reply:
That's fair since the `TODO` is there. I just hope we don't get too addicted to it and dig a hole. Works for me for now.